pkg/trace/config/config.go

```diff
@@ -210,6 +210,15 @@ func (c *AgentConfig) acquireHostname() error {
 // HTTPClient returns a new http.Client to be used for outgoing connections to the
 // Datadog API.
 func (c *AgentConfig) HTTPClient() *http.Client {
+	return &http.Client{
+		Timeout:   10 * time.Second,
+		Transport: c.HTTPTransport(),
+	}
+}
+
+// HTTPTransport returns a new http.Transport to be used for outgoing connections to
+// the Datadog API.
+func (c *AgentConfig) HTTPTransport() *http.Transport {
```

Comment:
I wonder if it'd be better to use the `New*` prefix on both these methods. Otherwise one may think it's a getter. Feel free to not apply this comment if it's too much to do (I'm unsure about all places used).